### ISSUE TITLE
Make SBDHBuilder#finishFields is public so it can be used directly

### ISSUE THREAD

**Issue by kukel** [phax/phase4#61]
[`SBDHBuilder#finishFields`](
https://github.com/phax/phase4/blob/301826859300caaeb906f2d0f3efd95f04bb8234/phase4-peppol-client/src/main/java/com/helger/phase4/peppol/Phase4PeppolSender.java#L1040) is protected but it is required when you want to process a already populated SBDH document. Since there is no real other reason to extend `Phase4PeppolSender`, making this public would be a nice thing to have.

**Comment by phax** [comment]
Well, I was looking at it and I am not conviced yet.
As the `finishFields` is called internally from `AbstractAS4MessageBuilder.sendMessage`, I don't see the reason why I should make this public. Can you please elaborate on the specific use case?

You can still derive from class `SBDHBuilder`  and extend the visibility of `finishFields` yourself. But I really miss the reason why someone from the outside would call this....

**Comment by kukel** [comment]
Because it failed sending if I did not explicitly call it. Will try to create a unit test to show.

**Comment by phax** [comment]
That is weird. That is done explicitly in a base class. Are you overriding some classes and not calling the `super`method?

**Comment by kukel** [comment]
Nope, working on a MainPhase4PeppolHelgerSBDH now ;-)... Few minutes

**Comment by kukel** [comment]
Took some time, our Test SMP underwent some unexpected maintenance. This issue is semi valid. 

When I use `.payload(bytes[])`, which I did, it fails and the `.finishFields()` is/seems needed  (and some more). Using  `.payloadAndMetadata(PeppolSBDHDocument) and marshalling the bytes/inputstream to `PeppolSBDHDocument` first. 

But this could be optimized I think since it is then made into bytes again. Why not pass on the bytes and in `.payload(bytes[])` marshall for the meta data AND set the bytes. Saves a relatively expensive operation.

**Comment by phax** [comment]
The original request (`public finishFields`) is obsolete — the class is now `PeppolUserMessageSBDHBuilder`, `finishFields` is invoked by `sendMessage()` before `isEveryRequiredFieldSet()`, and the metadata fields required alongside `payload(byte[])` are documented on the method. Keeping it protected.

The follow-up suggestion from the discussion is still open and I'll retitle the issue to it: add `payloadAndMetadata(byte[])` that parses the SBDH only to extract the metadata and forwards the original bytes unchanged, avoiding the parse --> regenerate round-trip (which is both wasteful and lossy for SBDH content not modelled by `PeppolSBDHData`).

**Comment by phax** [comment]
Finally fixed in 4.6.0
